### PR TITLE
IBX-673 Moving some jobs from Travis to GitHub actions in ezplatform-admin-ui

### PR DESCRIPTION
**Pull request by kacper-wieczorek-ibexa**
https://issues.ibexa.co/browse/IBX-673

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      - '[0-9]+.[0-9]+'
+  pull_request: ~
+
+env:
+  COMPOSER_ROOT_VERSION: 1.5.x-dev
```
Comment (Steveb-p) [inline .github/workflows/ci.yaml:11]: This should not be necessary and `branch-alias` should be used instead:

https://getcomposer.org/doc/articles/troubleshooting.md#dependencies-on-the-root-package

(afaik GH actions do not use shallow clones, or at the very least not shallow enough)

File: .travis.yml [hunk 2]
```diff
@@ -21,13 +21,13 @@ branches:
...
+    #    - name: "Code Style Check"
+    #      env: CHECK_CS=1
+    #    - name: "Unit tests"
+    #      env: PHPUNIT_CONFIG='phpunit.xml'
+    #    - name: "Unit tests (PHP 7.4)"
+    #      env: PHPUNIT_CONFIG='phpunit.xml'
+    #      php: 7.4
     - name: "Admin-UI tests on Clean P"
       env:
         - COMPOSE_FILE="doc/docker/base-dev.yml:doc/docker/selenium.yml"
```
Comment (Steveb-p) [inline .travis.yml:33]: By the way, apparently `docker-compose` is available (built-in) in GH actions @mnocon. So if you ever would like to transfer behat / selenium tests there as well, there is an option.

Comment (mnocon) [reply .travis.yml:33]: Yes, we plan to switch the selenium tests to Github Actions as well, but this will be a bigger task and I haven't investigated yet what's the best way to do it.

Good to know that docker-compose is available 👍

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (mnocon) [reply .github/workflows/ci.yaml:11]: The default fetch-depth is 1 (from https://github.com/actions/checkout), Composer was unable to resolve dependencies because of cyclic dependency between AdminUI and ez-support-tools and that's why it was added.

As far as I know Composer should create an implicit branch-alias for branch named like a version (1.5), if I understand correctly your suggestion we could try to add an explicit alias, something like: https://github.com/ezsystems/ezplatform-admin-ui/blob/1.5/composer.json#L46
```
        "branch-alias": {
            "dev-tmp_ci_branch": "1.5.x-dev",
            "dev-master": "1.5.x-dev",
            "dev-1.5": "1.5.x-dev"
        }
```

But I don't know if it works if the whole history is not fetched - definitely something we can check.

Comment (Steveb-p) [reply .github/workflows/ci.yaml:11]: Ok, didn't know that. If it's necessary for something, then :+1:

Although it makes it rather cumbersome to switch between versions, as you'd need to update both composer.json and CI declaration files. If any solution could be found that is contained within `composer.json` file, I think it should be prefered.

**Review by Steveb-p — APPROVED** [review]
:+1: despite `COMPOSER_ROOT_VERSION` usage.

As said in comment, if we can find something else, then great. If not... well, too bad.

File: .travis.yml [hunk 3]
```diff
@@ -21,13 +21,13 @@ branches:
...
 matrix:
   include:
-    - name: "Code Style Check"
-      env: CHECK_CS=1
-    - name: "Unit tests"
-      env: PHPUNIT_CONFIG='phpunit.xml'
-    - name: "Unit tests (PHP 7.4)"
-      env: PHPUNIT_CONFIG='phpunit.xml'
-      php: 7.4
+    #    - name: "Code Style Check"
```
Comment (mnocon) [inline .travis.yml:24]: Please remove commented code

File: composer.json [hunk 4]
```diff
@@ -37,13 +37,18 @@
...
+        "test": "phpunit -c phpunit.xml"
     },
     "extra": {
         "_ezplatform_branch_for_behat_tests": "2.5",
         "branch-alias": {
             "dev-tmp_ci_branch": "1.5.x-dev",
             "dev-master": "1.5.x-dev"
         }
+    },
+    "config": {
```
Comment (mnocon) [inline composer.json:51]: Why this change? IMHO we should not install with source by default.

File: .travis.yml [hunk 5]
```diff
@@ -21,13 +21,7 @@ branches:
...
 matrix:
   include:
-    - name: "Code Style Check"
-      env: CHECK_CS=1
-    - name: "Unit tests"
-      env: PHPUNIT_CONFIG='phpunit.xml'
-    - name: "Unit tests (PHP 7.4)"
-      env: PHPUNIT_CONFIG='phpunit.xml'
-      php: 7.4
+
```
Comment (mnocon) [inline .travis.yml:24]: Please remove empty line

File: .travis.yml [hunk 6]
```diff
@@ -55,18 +49,15 @@ before_install:
   - echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
 
 install:
-  # Install packages if needed
-  - if [ "${CHECK_CS}" == "1" -o "${PHPUNIT_CONFIG}" != "" ]; then travis_retry composer install --prefer-dist --no-interaction --no-suggest ; fi
-  # Prepare whole environment if needed
-  - if [ "${BEHAT_OPTS}" != "" ]; then ./.travis/prepare_ezplatform.sh ${INSTALL_EZ_INSTALL_TYPE}; fi
+
```
Comment (mnocon) [inline .travis.yml:52]: Please remove empty line

File: .travis.yml [hunk 7]
```diff
@@ -55,18 +49,15 @@ before_install:
...
 
 before_script:
-   - if [ "${SETUP_BEHAT_OPTS}" != "" ] ; then cd "$HOME/build/ezplatform"; docker-compose exec --user www-data app sh -c "bin/ezbehat $SETUP_BEHAT_OPTS" ; fi
+  - if [ "${SETUP_BEHAT_OPTS}" != "" ] ; then cd "$HOME/build/ezplatform"; docker-compose exec --user www-data app sh -c "bin/ezbehat $SETUP_BEHAT_OPTS" ;fi
 
 script:
-  - if [ "${CHECK_CS}" == "1" ] ; then ./vendor/bin/php-cs-fixer fix -v --dry-run --show-progress=estimating ; fi
-  - if [ "${PHPUNIT_CONFIG}" != '' ]; then ./vendor/bin/phpunit -c "${PHPUNIT_CONFIG}"; fi
-  - if [ "${BEHAT_OPTS}" != "" ]; then cd "$HOME/build/ezplatform"; docker-compose exec --user www-data app sh -c "bin/ezbehat ${BEHAT_OPTS}" ; fi
+
```
Comment (mnocon) [inline .travis.yml:59]: Please remove empty line

File: .travis.yml [hunk 8]
```diff
@@ -77,4 +68,5 @@ after_failure:
   - docker ps -s
 
 after_script:
-  - if [ "${BEHAT_OPTS}" != "" ] ; then bin/ezreport ; fi
+
```
Comment (mnocon) [inline .travis.yml:71]: Please remove empty line

File: composer.json [hunk 9]
```diff
@@ -46,4 +48,5 @@
             "dev-master": "1.5.x-dev"
         }
     }
+
```
Comment (mnocon) [inline composer.json:51]: Please remove empty line

**Review by mnocon — COMMENTED** [review]
Looks good, just some nitpicks and we're good to go 😄

**Review by mnocon — APPROVED** [review]
Looks good, thanks!

**Review by micszo — APPROVED** [review]
